draft-ietf-quic-transport.md

```diff
@@ -3838,143 +3838,135 @@ The details of loss detection and congestion control are described in
-with ECN markings.  To provide robust connectivity in the presence of such
-devices, each endpoint independently validates ECN counts and disables ECN if
-errors are detected.
+For example, if one each of an Initial, 0-RTT, Handshake, and 1-RTT QUIC packet
+are coalesced, the corresponding counts for the Initial and Handshake packet
+number space will be incremented by one and the counts for the application data
+packet number space will be increased by two.
 
-Endpoints validate ECN for packets sent on each network path independently.  An
-endpoint thus validates ECN on new connection establishment, when switching to a
-server's preferred address, and on active connection migration to a new path.
-{{ecn-alg}} describes one possible algorithm for testing paths for ECN support.
+An ECN count is not incremented when the received IP packet does not results in
+a QUIC packets being processed.  A QUIC packet detected by a receiver as a
+duplicate also does not increase an ECN count; see {{security-ecn}} for relevant
+security concerns.
 
-Even if an endpoint does not use ECN markings on packets it transmits, the
-endpoint MUST provide feedback about ECN markings received from the peer if they
-are accessible.  Failing to report ECN counts will cause the peer to disable ECN
-marking.
 
+### ECN Validation {#ecn-validation}
 
-#### Sending ECN Markings
+It is possible for faulty network devices to corrupt or erroneously drop packets
+that set an ECN codepoint.  To provide robust connectivity in the presence of
+such devices, an endpoint validates the ECN counts for each network path  and
+disables use of ECN on that path if errors are detected.
 
-To start ECN validation, an endpoint SHOULD do the following when sending
-packets on a new path to a peer:
+To perform ECN validation for a new path:
 
-* Set the ECT(0) codepoint in the IP header of early outgoing packets sent on a
-  new path to the peer ({{!RFC8311}}).
+* The endpoint SHOULD set the ECT(0) codepoint in the IP header of early
+  outgoing packets sent on a new path to the peer ({{!RFC8311}}).
 
-* If all packets that were sent with the ECT(0) codepoint are eventually deemed
-  lost (Section 6 of {{QUIC-RECOVERY}}), validation is deemed to have failed.
+* The endpoint SHOULD monitor whether all packets sent with an ECT codepoint are
+  eventually deemed lost (Section 6 of {{QUIC-RECOVERY}}), deeming this as an
+  indication that ECN validation has failed.
 
-To reduce the chances of misinterpreting congestive loss as packets dropped by a
-faulty network element, an endpoint could set the ECT(0) codepoint for only the
-first ten outgoing packets on a path, or for a period of three RTTs, whichever
-occurs first.
+To reduce the chances of misinterpreting loss of packets dropped by a faulty
+network element, an endpoint could set an ECT codepoint for only the first ten
+outgoing packets on a path, or for a period of three RTTs, whichever occurs
+first.
```

Comment:
More needs to be said here about why this might be beneficial.

Reply:
This mechanism was there already, so it wasn't my idea. When I edited,  I didn't see where it did hard - nor did I see benefit - so is it needed?

Reply:
@mirjak - What do you suggest (remove this text)?

Reply:
I argued in another discussion already that we should just recommend to mark all packets as ECT(0) until validation fails as this case where packet get dropped is basically not existence anymore. However, this was not accepted as people felt this approach would not increase implementation complexity. I still think the text should note that this is a really rare case and I had an editorial PR for that that was also not expected, so I would propose anything again. Just wanted to state my opinion...

Reply:
I think we all need to find text we can agree upon. Here is a proposal:

"If an endpoint has cause to expect that packets setting an ECT codepoint are being dropped by faulty network element, the endpoint could set an ECT codepoint for only the first ten outgoing packets on a path, or for a period of three RTTs. If an endpoint then discovers all non-zero ECN codepoints fail to be delivered, it can assume the path did not support ECN.

Other methods of probing paths for ECN support are possible,..."

Reply:
Looks good to me!

Reply:
I don't think your rewording did change anything from the original text.

Reply:
Sorry, I think your wording is fine. I looked at this text early in the same section which still seems to indicate that dropping is a real problem:

```
It is possible for faulty network devices to corrupt or erroneously drop
 packets that carry a non-zero ECN codepoint. To ensure connectivity in the
 presence of such devices, an endpoint validates the ECN counts for each network
 path and disables use of ECN on that path if errors are detected.
```

Reply:
I agree,  and suggest this sentence can now be removed, and is now covered by the clarified later sentence: "It is possible for faulty network devices to corrupt or erroneously drop packets that carry a non-zero ECN codepoint."